sdk/core/Azure.Core/tests/public/PatchModels/ChildPatchModel.Serialization.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Text.Json;
+using Azure.Core.Json;
+using Azure.Core.Serialization;
+
+namespace Azure.Core.Tests.PatchModels
+{
+    public partial class ChildPatchModel : IModelJsonSerializable<ChildPatchModel>
+    {
+        ChildPatchModel IModelJsonSerializable<ChildPatchModel>.Deserialize(ref Utf8JsonReader reader, ModelSerializerOptions options)
+        {
+            MutableJsonDocument mdoc = MutableJsonDocument.Parse(ref reader);
+            return new ChildPatchModel(mdoc.RootElement);
+        }
+
+        ChildPatchModel IModelSerializable<ChildPatchModel>.Deserialize(BinaryData data, ModelSerializerOptions options)
+        {
+            MutableJsonDocument mdoc = MutableJsonDocument.Parse(data);
+            return new ChildPatchModel(mdoc.RootElement);
+        }
+
+        void IModelJsonSerializable<ChildPatchModel>.Serialize(Utf8JsonWriter writer, ModelSerializerOptions options)
+        {
+            _element.WriteTo(writer, options.Format.ToString());
+        }
+
+        BinaryData IModelSerializable<ChildPatchModel>.Serialize(ModelSerializerOptions options)
+        {
+            using MemoryStream stream = new();
```

Comment:
you can follow the [implementation here](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/tests/common/ModelSerializationTests/Models/DiscriminatorSet/ModelX.cs#L137-L142).